src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/resources/viewmodel/ResourcesViewModel.kt

```diff
@@ -1,43 +1,48 @@
 package org.wycliffeassociates.otter.jvm.app.ui.resources.viewmodel
 
 import javafx.application.Platform
-import javafx.beans.property.SimpleObjectProperty
-import javafx.beans.property.SimpleStringProperty
 import org.wycliffeassociates.otter.common.data.workbook.*
+import org.wycliffeassociates.otter.common.domain.content.Recordable
 import org.wycliffeassociates.otter.common.utils.mapNotNull
+import org.wycliffeassociates.otter.jvm.app.ui.resourcetakes.viewmodel.TakesViewModel
+import org.wycliffeassociates.otter.jvm.app.ui.workbook.viewmodel.WorkbookViewModel
 import org.wycliffeassociates.otter.jvm.app.widgets.resourcecard.model.ResourceGroupCardItemList
 import org.wycliffeassociates.otter.jvm.app.widgets.resourcecard.model.resourceGroupCardItem
 import tornadofx.*
 
 class ResourcesViewModel : ViewModel() {
-    val activeWorkbookProperty = SimpleObjectProperty<Workbook>()
-    val workbook: Workbook
-        get() = activeWorkbookProperty.value
+    private val takesViewModel: TakesViewModel by inject()
+    private val workbookViewModel: WorkbookViewModel by inject()
 
-    val activeChapterProperty = SimpleObjectProperty<Chapter>()
-    val chapter: Chapter
-        get() = activeChapterProperty.value
-
-    val activeResourceSlugProperty = SimpleStringProperty()
-    val resourceSlug: String
-        get() = activeResourceSlugProperty.value
-
-    val resourceGroups: ResourceGroupCardItemList = ResourceGroupCardItemList(mutableListOf())
+    val resourceGroups: ResourceGroupCardItemList = ResourceGroupCardItemList()
 
     fun loadResourceGroups() {
-        chapter
-            .children
-            .startWith(chapter)
-            .mapNotNull { resourceGroupCardItem(it, resourceSlug, onSelect = this::navigateToTakesPage) }
-            .buffer(2) // Buffering by 2 prevents the list UI from jumping while groups are loading
-            .subscribe {
-                Platform.runLater {
-                    resourceGroups.addAll(it)
+        workbookViewModel.chapter.let { chapter ->
```

Comment:
why use let here?

Reply:
I use `chapter` again at line 23 so I didn't want to write `workbookViewModel.chapter` again, but maybe I should have just done `val chapter = workbookViewModel.chapter` at the start